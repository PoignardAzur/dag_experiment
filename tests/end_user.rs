use dag_experiment::{Graph, GraphCache};
// TODO
#[allow(unused)]
use dag_experiment::{Operable, Operation};

// TODO - Implement custom operation

#[test]
fn basic_tree() {
    let mut graph = Graph::new();
    let mut cache = GraphCache::new();
    let id_1 = graph.add_op(Operation::Leaf(42.0));
    let id_2 = graph.add_op(Operation::Leaf(10.0));
    let id_sum = graph.add_op(Operation::Sum(id_1, id_2));
    let id_3 = graph.add_op(Operation::Leaf(2.0));
    let id_product = graph.add_op(Operation::Product(id_sum, id_3));

    assert_eq!(graph.compute_from_root(&mut cache, id_product), 104.0);
}
