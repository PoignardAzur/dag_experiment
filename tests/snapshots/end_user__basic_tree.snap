---
source: tests/end_user.rs
assertion_line: 41
expression: graph.get_debug_tree(id_product)
---
x, y -> x * 10 + y * 3
| x, y -> x + y
| | 42
| | 10
| 2
